cpp/src/slice2cs/Gen.cpp

```diff
@@ -2397,6 +2397,7 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& operation)
     string name = fixId(opName);
     string asyncName = opName + "Async";
     string internalName = "_iceI_" + opName + "Async";
+    bool oneway = cl->hasMetaData("oneway") || operation->hasMetaData("oneway");
```

Comment:
Is there any reason why we shouldn't let users declare entire oneway interfaces? It's somewhat common that we'll let metadata be inherited from containers. I know we at least have the same logic for 'preserve-slice' and 'amd', but probably others.

Reply:
`oneway` applies only to void operations, so this would restrict an interface to void operations, which I'd find odd and unnecessary.

Reply:
I agree it's probably not a common thing to want, but I don't see a reason not to allow it.
And it isn't like it adds any complexity to our code, so why not allow users the extra flexibility?

Pretend you have a helper interface that sends stringified data to a server, but since there's no templates in Slice, there's loads of operations, one for each data-type. *shrugs* It's safer and easier to mark the whole interface with metadata instead of every single operation.

Reply:
It would add complexity:
 - is this metadata inherited by derived classes?
 - is there an opposite metadata to disable its effect on a single operation?

So please make it operation-only, just like idempotent (a keyword) is operation-only.

Reply:
I presume you mean derived interfaces?
It doesn't seem like we document it anywhere, but metadata isn't inherited. But this isn't _added_ complexity, metadata already works like this, like 'amd' and 'preserve-slice' as I mentioned before.

No, but again, I don't think lacking a 'not-oneway' metadata is adding any complexity. If you have an interface where not all operations should be oneway, then I think it's pretty clear not to attach oneway to the entire interface...

If you still want it operation only, I'll do it, but I really don't think it's a reasonable restriction, at least not for the reasons you gave. And with these limited semantics, I think oneway should of been done as a keyword like idempotent, and not metadata.

Reply:
As explained on the corresponding issue #747, I believe metadata is more appropriate. And yes, please make it operation-only.

Reply:
Alright, I've fixed it all up if you wanna take another look through!